packages/ComponentLibrary/src/components/PrimaryTab/index.tsx

```diff
@@ -0,0 +1,130 @@
+import React, { useState } from 'react';
+import {
+  Tabs,
+  Tab,
+  Box,
+  Menu,
+  MenuItem,
+  ListItemIcon,
+  Tooltip,
+} from '@mui/material';
+import CheckIcon from '@mui/icons-material/Check';
+import { PrimaryTabsProps, TabItem } from './types';
+import { menuStyle, styles, sx, tabIndicatorProps } from './styles';
+import IconButton from '../IconButton';
+
+const PrimaryTabs: React.FC<PrimaryTabsProps> = ({ tabs, onChange, icon }) => {
+  const [selectedTab, setSelectedTab] = useState(tabs[0]?.id || '');
+  const [hoveredTab, setHoveredTab] = useState<string | null>(null);
+  const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
+
+  const handleChange = (_: React.SyntheticEvent, newValue: string) => {
```

Comment:
You should wrap all the callbacks used in the childs with useCallback.
useCallback is a React Hook that lets you cache a function definition between re-renders.

When you optimize rendering performance, you will sometimes need to cache the functions that you pass to child components.
To cache a function between re-renders of your component, wrap its definition into the useCallback hook.

example usage:
https://react.dev/reference/react/useCallback#skipping-re-rendering-of-components

https://react.dev/reference/react/useCallback#usecallback